src/hooks/useOwnedSafes.ts

```diff
@@ -16,26 +16,23 @@ type OwnedSafesCache = {
 }
 
 const useOwnedSafes = (): OwnedSafesCache['walletAddress'] => {
-  const { safe } = useSafeInfo()
+  const chainId = useChainId()
   const { address: walletAddress } = useWallet() || {}
   const [ownedSafesCache, setOwnedSafesCache] = useLocalStorage<OwnedSafesCache>(CACHE_KEY)
 
-  const [ownedSafes, error] = useAsync<OwnedSafes>(() => {
-    if (!safe.chainId || !walletAddress) return
-    return getOwnedSafes(safe.chainId, walletAddress)
-  }, [safe.chainId, walletAddress])
+  const [_, error] = useAsync<void>(async () => {
+    if (!chainId || !walletAddress) return
 
-  useEffect(() => {
-    if (!ownedSafes || !walletAddress || !safe.chainId) return
+    const ownedSafes = await getOwnedSafes(chainId, walletAddress)
 
     setOwnedSafesCache((prev) => ({
```

Comment:
Setting the state inside useAsync is IMHO an anti-pattern.
Is there another way to fix it?

Reply:
What's the root cause of the bug? The chainId changes over time?

Reply:
If you want to keep it in this imperative style, I would suggest just using an effect like so:
```
useEffect() => {
  getOwnedSafes(chainId).then(data => setOwnedSafesCache(prev => {
    ...
  }))
}, [chainId, etc])
```

Reply:
The root cause is that the `Drawer` unmounts when switching to a safe on a different network but there is a brief period before it unmounts where the `chainId` already changed but the `ownedSafes` data is stale. It runs the `useEffect` hook which updates the local storage for the new `chainId` with the data from the previous network.

Alternatively, we could keep the `Drawer` mounted through the `keepMounted` prop that MUI offers which solves the issue but @iamacook raised a valid point that this would not update a newly created safe unless the user either reconnects their wallet or switches to a different chain.

Another thing we could do is to not use `useAsync` for this case to avoid this anti-pattern.